Vagrantfile

```diff
@@ -1,71 +1,302 @@
+
+    clone_crystal_from_vagrant(c)
   end
+end
+
+# define a ubuntu box with the given *name*
+# *llvm* values: 6.0, 7, (empty), Hash with {url:, path:}
+def define_debian(config, name:, dist:, bits:, llvm: nil, github_targz: INSTALL_GITHUB_TARGZ)
+  config.vm.define(name) do |c|
+    c.vm.box = "debian/#{dist}#{bits}"
 
-  config.vm.define "freebsd" do |c|
-    c.ssh.shell = "csh"
-    c.vm.box = "freebsd/FreeBSD-10.2-RELEASE"
+    setup_memory(config, bits)
+
+    c.vm.provision :shell, inline: %(
+      echo '' > /etc/profile.d/crystal.sh
+
+      apt-get install -y software-properties-common apt-transport-https dirmngr curl build-essential git pkg-config
+      apt-get install -y zlib1g-dev libxml2-dev libyaml-dev libreadline-dev libgmp3-dev libssl-dev libpcre3-dev \
+        libevent-dev libevent-core-2.0-5 libevent-extra-2.0-5 libevent-openssl-2.0-5 libevent-pthreads-2.0-5
+
+      #{install_llvm(dist, llvm)}
+    )
+
+    install_crystal(c, "debian", dist: dist, bits: bits, github_targz: github_targz)
+
+    clone_crystal_from_vagrant(c)
+  end
+end
+
+def define_freebsd(config, name:, box:)
+  config.vm.define name do |c|
+    c.ssh.shell = "sh"
+    c.vm.box = "freebsd/FreeBSD-#{box}"
+
+    c.vm.hostname = name
+    c.vm.base_mac = "6658695E16F0"
 
     c.vm.network "private_network", type: "dhcp"
     c.vm.synced_folder ".", "/vagrant", type: "nfs"
 
-    # to build boehm-gc from git repository:
-    #c.vm.provision :shell, inline: %(
-    #  pkg install -y libtool automake autoconf libatomic_ops
-    #)
-
     c.vm.provision :shell, inline: %(
-      pkg install -y git gmake pkgconf pcre libunwind clang36 libyaml gmp libevent2
+      mkdir -p /etc/profile.d
+      echo '' > /etc/profile.d/crystal.sh
+      echo '. /etc/profile.d/crystal.sh' >> /etc/profile
+
+      pkg install -y git gmake pkgconf
+      pkg install -y libyaml gmp libunwind libevent pcre boehm-gc-threaded
+      pkg install -y llvm60
     )
 
-    clone_crystal_from_vagrant.call(c)
+    install_crystal(c, "freebsd", dist: nil, bits: 64, github_targz: false)
+
+    clone_crystal_from_vagrant(c)
   end
+end
 
-  config.vm.provider "virtualbox" do |vb|
-    vb.memory = 4096
-    vb.cpus = 2
+def define_alpine(config, name:, bits:)
+  config.vm.define name do |c|
+    c.vm.box = "alpine/alpine#{bits}"
+
+    setup_memory(config, bits)
+
+    c.vm.provision :shell, inline: %(
+      echo '' > /etc/profile.d/crystal.sh
+
+      echo 'export LLVM_CONFIG=llvm4-config' >> /etc/profile.d/crystal.sh
+      echo 'export FLAGS="--target x86_64-linux-musl --link-flags=-no-pie --static"' >> /etc/profile.d/crystal.sh
+
+      apk add --no-cache \
+        tar curl git gcc g++ make automake libtool autoconf bash coreutils paxmark \
+        zlib-dev yaml-dev pcre-dev libxml2-dev gmp-dev readline-dev libressl-dev libatomic_ops libevent-dev gc-dev \
+        llvm4-dev llvm4-static || echo "WARNING: apk failed (ignored)"
```

Comment:
Why only LLVM 4?

Reply:
Is the one used in the build. But I added `llvm:` (default 4) argument and can be used with 5 to define another alpine machine.